src/InstrumentationEngine.Lib/Logging.cpp

```diff
@@ -217,7 +217,28 @@ void CLogging::XmlDumpHelper::WriteStringNode(const WCHAR* name, const WCHAR* va
     {
         return;
     }
-    m_result << m_childrenIndent << _T("<") << name << _T(">") << value << _T("</") << name << _T(">") << _T("\r\n");
+    m_result += m_childrenIndent + _T("<") + name + _T(">") + value + _T("</") + name + _T(">") + _T("\r\n");
+}
+
+namespace {
+    tstring format_hex(ULONG32 value, size_t width = 10)
+    {
+        tstring result(width, _T('0'));
+        result[1] = _T('x');
+
+        size_t pos = width - 1;
+        while (value != 0) {
+            const auto nibble = value & 0xf;
+            if (nibble < 0xa)
+                result[pos] = _T('0') + nibble;
+            else
+                result[pos] = _T('a') + nibble - 0xa;
+            --pos;
```

Comment:
>--pos [](http://example.com/codeflow?start=12&length=5)

If width width isn't large enough to contain all of the digits, then pos will underflow, potentially causing a buffer overrun (std::basic_string::operator[] does not do bounds checking)